pyart/aux_io/arm_vpt.py

```diff
@@ -0,0 +1,351 @@
+        if field_name is None:
+            if exclude_fields is not None and key in exclude_fields:
+                continue
+            field_name = key
+        fields[field_name] = cfradial._ncvar_to_dict(ncvars[key])
+
+    if 'ray_n_gates' in ncvars:
+        shape = (len(ncvars['time']), len(ncvars['range']))
+        ray_n_gates = ncvars['ray_n_gates'][:]
+        ray_start_index = ncvars['ray_start_index'][:]
+        for dic in fields.values():
+            cfradial._unpack_variable_gate_field_dic(
+                dic, shape, ray_n_gates, ray_start_index)
+
+    # 4.5 instrument_parameters sub-convention -> instrument_parameters dict
+    # this section needed multiple changes and/or additions since the
+    # instrument parameters were primarily located in the global attributes
+    # this section is likely still incomplete
+    omega = float(ncobj.radar_operating_frequency.split()[0])
+    frequency = {
+        'data': np.array([omega / 1e9], dtype=np.float32),
+        'long_name': 'Radar operating frequency',
+        'standard_name': 'radiation_frequency',
+        'units': 'Hz',
+        'meta_group': 'instrument_parameters',
+    }
+    prt_mode = {
+        'data': np.array(['fixed'], dtype=np.str),
+        'long_name': 'Pulsing mode',
+        'standard_name': 'transmit_pulse_mode',
+        'units': 'unitless',
+        'meta_group': 'instrument_parameters',
+        'comment': ('Options are "fixed", "staggered", "dual". Assumed '
+                    '"fixed" if missing'),
+    }
+    prf = float(ncobj.pulse_repetition_frequency.split()[0])
+    prt = {
+        'data': (1.0 / prf) * np.ones(ncvars['time'].size, dtype=np.float32),
+        'long_name': 'Pulse repetition time',
+        'standard_name': 'pulse_repetition_time',
+        'units': 'seconds',
+        'meta_group': 'instrument_parameters',
+        'comments': 'For staggered PRT, also see prt_ratio',
+    }
+    v_nq = float(ncobj.nyquist_velocity.split()[0])
+    nyquist_velocity = {
+        'data': v_nq * np.ones(ncvars['time'].size, dtype=np.float32),
+        'long_name': 'Nyquist velocity',
+        'standard_name': 'unambiguous_doppler_velocity',
+        'units': 'meters_per_second',
+        'meta_group': 'instrument_parameters',
+    }
+    samples = int(ncobj.num_spectral_averages)
+    n_samples = {
+        'data': samples * np.ones(ncvars['time'].size, dtype=np.int32),
+        'long_name': 'Number of samples used to compute moments',
+        'standard_name': 'number_of_samples_used_to_compute_moments',
+        'units': 'unitless',
+        'meta_group': 'instrument_parameters',
+    }
+
+    # 4.6 radar_parameters sub-convention -> instrument_parameters dict
+    # this section needed multiple changes and/or additions since the
+    # radar instrument parameters were primarily located in the global
+    # attributes
+    # this section is likely still incomplete
+
+    instrument_parameters = {
```

Comment:
If a try/if block is used above then instrument_parameters should be built up sequentially with a check for an empty dictionary at the end.